test/k23si/3SITxnTest.cpp

```diff
@@ -1794,10 +1797,690 @@ seastar::future<> testScenario02() {
+                .timestamp = {(ts.tEndTSECount() - 10000), 123, 1000}, 
+                .priority = dto::TxnPriority::High },
+            dto::K23SI_MTR { // newMtr: txn with later ts, same priority, same tso id
+                .txnid = txnids++, 
+                .timestamp = {(ts.tEndTSECount() + 10000), 123, 1000}, 
+                .priority = dto::TxnPriority::High },
+            dto::K23SI_MTR { // tsoSmMtr: txn with same ts, same priority, smaller tso id
+                .txnid = txnids++, 
+                .timestamp = {(ts.tEndTSECount() + 10000), 100, 1000}, 
+                .priority = dto::TxnPriority::High },
+            dto::K23SI_MTR { // tsoBgMtr: txn with same ts, same priority, bigger tso id
+                .txnid = txnids++, 
+                .timestamp = {(ts.tEndTSECount() + 10000), 200, 1000}, 
+                .priority = dto::TxnPriority::High },
+            dto::K23SI_MTR {.txnid = txnids++, .timestamp = ts, .priority = dto::TxnPriority::Medium}, // incumbent mtr
+            dto::Key {.schemaName = "schema", .partitionKey = "SC05_pkey1", .rangeKey = "rKey1"},
+            DataRec {.f1="SC05_f1_one", .f2="SC04_f2_one"},
+            DataRec {.f1="SC05_f1_zero", .f2="SC04_f2_zero"},
+            [&](auto& fakeCfxMtr, auto& prioHighMtr, auto& prioLowMtr, auto& oldMtr, auto& newMtr, auto& tsoSmMtr, auto& tsoBgMtr, \
+                auto& incMtr, auto& k1, auto& v1, auto& v0) {
+                return doWrite(k1, v1, incMtr, k1, collname, false, true, ErrorCaseOpt::NoInjection)
+                .then([](auto&& response) {
+                    auto& [status, resp] = response;
+                    K2EXPECT(status, dto::K23SIStatus::Created);
+                })
+                .then([&] {
+                    K2INFO("------- SC05.case1 (fake-read-write conflict) -------");
+                    return doRead(k1, fakeCfxMtr, collname, ErrorCaseOpt::NoInjection)
+                    .then([&](auto&& response) {
+                        auto& [status, val] = response;
+                        K2INFO("SC05.case01::OP_read_earlier. " << "status: " << status.code << " with MESG: " << status.message);
+                        K2INFO("Value of k1 in fakeCfxMtr: " << val);
+                        K2EXPECT(status, dto::K23SIStatus::OK);
+                        K2EXPECT(val, v0);
+                    });
+                })
+                .then([&] {
+                    K2INFO("------- SC05.case2 (Txn with higher priority encounters a WI) -------");
+                    return doWrite(k1, v1, prioHighMtr, k1, collname, false, true, ErrorCaseOpt::NoInjection)
+                    .then([](auto&& response) {
+                        auto& [status, resp] = response;
+                        K2INFO("SC05.case02::OP_push_prioHighMtr. " << "status: " << status.code << " with MESG: " << status.message);
+                        K2EXPECT(status, dto::K23SIStatus::Created);
+                    });
+                })
```

Comment:
Same for these tests, we can use the K23SIInsepect RPCs to check that that the aborted transaction is in the force aborted state.